src/agents/agent.py

```diff
@@ -63,7 +64,16 @@ def search_queries(self, queries: list, project_name: str) -> dict:
         results = {}
         
         knowledge_base = KnowledgeBase()
-        bing_search = BingSearch()
+        user_search_engine = Config().get_search_engine()
+
+        if user_search_engine == "DDGS":
+            search_engine = DuckDuckGoSearch()
+        else:
+            """
+            Set default search engine to Bing
+            """
```

Comment:
**Scalability Issue**: The dynamic assignment of search_engine based on user preferences introduces a potential scalability issue by not considering the instantiation cost or the resource utilization of different search engines, which might vary significantly. <br> **Fix**: Implement a search engine factory that caches instances of search engines and reuses them. This approach minimizes the instantiation cost and manages resource utilization effectively, enhancing scalability. <br> **Code Suggestion**: 
 ```
 @@ -66,10 +66,15 @@
 -        user_search_engine = Config().get_search_engine()
 -        if user_search_engine == "DDGS":
 -            search_engine = DuckDuckGoSearch()
 -        else:
 -            search_engine = BingSearch()
 +        search_engine = SearchEngineFactory.get_engine(Config().get_search_engine())
 ```